### ISSUE TITLE
Yaml encodes multiline strings ugly

### ISSUE THREAD

**Issue by rosik** [tarantool/tarantool#6304]
**Bug description**

Yaml encoder produces ugly output:

```text
tarantool> return {foo = 'a\nb\nc\n'}
---
- foo: 'a

    b

    c

    '
...

```

But it can do better in case the string contains double new line:

```text
tarantool> return {foo = 'a\nb\n\nc\n'}
---
- foo: |
    a
    b

    c
...

```

Let's always encode multiline strings using yaml block syntax `|`.

**Comment by Totktonada** [comment]
Related to #3012.

**Comment by Lord-KA** [comment]
Seems to be fixed by #7250.

**Comment by Lord-KA** [comment]
Fixed by [yaml_pretty_multiline](https://github.com/tarantool/tarantool/wiki/compat:yaml_pretty_multiline) option in tarantool.compat:

```
Tarantool 2.10.0-beta1-1835-g2ba2c6370
type 'help' for interactive help
tarantool> c = require('compat')
---
...

tarantool> c
---
- - yaml_pretty_multiline: default (old)
  - sql_seq_scan_default: default (old)
  - json_escape_forward_slash: default (old)
  - fiber_channel_close_mode: default (old)
  - fiber_slice_default: default (old)
  - box_cfg_replication_sync_timeout: default (old)
...

tarantool> c.yaml_pretty_multiline = 'new'
---
...

tarantool> return {foo = 'a\nb\nc\n'}
---
- foo: |
    a
    b
    c
...

tarantool> 
```
